src/modules-lua/noit/module/resmon/consul.lua

```diff
@@ -0,0 +1,388 @@
+</module>
+]=]);
+  return 0
+end
+
+-- http://lua-users.org/wiki/SplitJoin
+function split(str, delim, maxNb)
+  -- Eliminate bad cases...
+  if string.find(str, delim) == nil then
+    return { str }
+  end
+  if maxNb == nil or maxNb < 1 then
+    maxNb = 0    -- No limit
+  end
+  local result = {}
+  local pat = "(.-)" .. delim .. "()"
+  local nb = 0
+  local lastPos
+  for part, pos in string.gfind(str, pat) do
+    nb = nb + 1
+    result[nb] = part
+    lastPos = pos
+    if nb == maxNb then
+      break
+    end
+  end
+  -- Handle the last field
+  if nb ~= maxNb then
+    result[nb + 1] = string.sub(str, lastPos)
+  end
+  return result
+end
+
+function escape(s)
+  s = string.gsub(s, "([%%,:/&=+%c])", function (c)
+                    return string.format("%%%02X", string.byte(c))
+  end)
+  s = string.gsub(s, " ", "+")
+  return s
+end
+
+function fix_config(inconfig)
+  local config = {}
+
+  for k,v in pairs(inconfig) do config[k] = v end
+
+  if not config.url then
+    config.url = 'http:///v1/health/state/any'
+  end
+  if not config.port then
+    config.port = 8500
+  end
+
+  if config.consul_dc ~= nil then
+    config.url = config.url .. '?dc=' escape(config.consul_dc)
+  end
+
+  return config
+end
+
+function set_check_metric(check, name, type, value)
+    if type == 'i' then
+        check.metric_int32(name, value)
+    elseif type == 'I' then
+        check.metric_uint32(name, value)
+    elseif type == 'l' then
+        check.metric_int64(name, value)
+    elseif type == 'L' then
+        check.metric_uint64(name, value)
+    elseif type == 'n' then
+        check.metric_double(name, value)
+    elseif type == 's' then
+        check.metric_string(name, value)
+    else
+        check.metric(name, value)
+    end
+end
+
+-- look for a "Status" key in o and add to count_table for that status
+function count_status(o, count_table)
+   for k, v in pairs(o) do
+      if count_table ~= nil and k == "Status" then
+         if count_table[v] == nil then
+            count_table[v] = 1
+         else
+            count_table[v] = count_table[v] + 1
+         end
+      end
+   end
+end
+
+function json_metric(check, prefix, o, index)
+   local cnt = 0
+   if type(o) == "table" then
+      for k, v in pairs(o) do
+         local np
+         if type(v) ~= "table" and prefix ~= nil then
+            np = prefix .. '`' .. k
+            set_check_metric(check, np, string.find(k, "Index") and 'L' or 's', v)
+            cnt = cnt + 1
+         end
+      end
+      return cnt
+   end
+   return cnt
+end
+
+function make_safe(s)
+   local x, y = string.gsub(s, ' ', '_')
+   x, y = string.gsub(x, '[^A-Za-z0-9_-]', '')
+   return x
+end
+
+function first_to_upper(str)
+    return (str:gsub("^%l", string.upper))
+end
+
+function json_to_metrics(check, doc, blacklists)
+    local services = 0
+    check.available()
+    local data = doc:document()
+    if data ~= nil then
+       if string.find(check.config.url, "/health/node") then
```

Comment:
I was poorly attempting future compatibility (Assuming they don't change the name of the endpoints in `v2`.. This seems like a stupid idea.. will fix.